Strip trailing spaces in server messages

citrin opened the pull request:
Add workaround to fix freenode SASL auth.
At least one freende server (weber.freenode.net) adds a space to the reply:

    2019/02/28 17:42:40 <-- ":weber.freenode.net CAP * ACK :account-notify \r\n"
    2019/02/28 17:42:40 <-- ":weber.freenode.net CAP * ACK :sasl \r\n"

It looks like a server bug, but other IRC clients ignore trailing spaces, and I think it is a good idea to ignore them in go-ircevent too.

citrin committed:
Strip trailing spaces in server messages

Add workaround to fix freenode SASL auth.
At least one freende server (weber.freenode.net) adds a space to the reply:

    2019/02/28 17:42:40 <-- ":weber.freenode.net CAP * ACK :account-notify \r\n"
    2019/02/28 17:42:40 <-- ":weber.freenode.net CAP * ACK :sasl \r\n"

jwheare commented:
Fwiw this is a valid enough server response. Capabilities are usually requested and acked more than one at a time, and the spec says to split the final arg on spaces before processing.

citrin commented:
Then may be another fix is needed. For example final arg can be split before saving in Event.Arguments. But this will require changes in other parts of code (and if irc.Event is exposed in API it will break go-ircevent users).

To strip trailing spaces is an easy fix and it should not break anything.

More info about my issue is here: https://github.com/irccloud/irccat/pull/11

citrin commented:
One more option - to strip spaces in a callback added by setupSASLCallbacks, but I'm not sure that it will be better.

citrin commented:
And one more option - strip spaces only around parameter list (reply part after ":"):
```
--- a/irc.go
+++ b/irc.go
@@ -147,7 +147,7 @@ func parseToEvent(msg string) (*Event, error) {
 	event.Code = strings.ToUpper(args[0])
 	event.Arguments = args[1:]
 	if len(split) > 1 {
-		event.Arguments = append(event.Arguments, split[1])
+		event.Arguments = append(event.Arguments, strings.Trim(split[1], " "))
 	}
 	return event, nil
 
@@ -534,7 +534,7 @@ func (irc *Connection) negotiateCaps() error {
 				cap_chan <- true
 			}
 		} else if command == "ACK" || command == "NAK" {
-			for _, cap_name := range strings.Split(strings.TrimSpace(e.Arguments[2]), " ") {
+			for _, cap_name := range strings.Split(e.Arguments[2], " ") {
 				if cap_name == "" {
 					continue
 				}
```

russss commented:
I've fixed this in a nicer way in thoj/go-ircevent#115.
